REFERENCE.md

```diff
@@ -0,0 +1,233 @@
+
+Default value: `undef`
+
+##### `target`
+
+Data type: `Any`
+
+Specifies the absolute path to the base location where you want to install IBM Installation Manager. Defaults to `/opt/IBM/InstallationManager`.
+
+Default value: `undef`
+
+##### `manage_user`
+
+Data type: `Any`
+
+Whether or not to manage the user that will be installing the IBM IM. Default: `false`.
+
+Default value: `false`
+
+##### `user`
+
+Data type: `Any`
+
+Specifies the user to run the installation as. Defaults to `root`. Note that installing as a different user might cause undefined behavior. Consult IBM's documentation for details.
+Note that installing as a user other than `root` might result in undefined behavior. Consult IBM's documentation for details. Installations by a non-root user won't share installation data with the rest of the system.
+
+Default value: 'root'
+
+##### `user_home`
+
+Data type: `Any`
+
+Specifies the home directory for the specified user. Required if you're installing in a mode other than 'administrator'.
+
+Default value: `undef`
+
+##### `manage_group`
+
+Data type: `Any`
+
+When in 'group' mode, whether or not to manage the group that will be installing the IBM IM. Default: `false`.
+
+Default value: `false`
+
+##### `group`
+
+Data type: `Any`
+
+Specifies the group to run the installation as. Defaults to `root`.
+Note that installing as a user other than `root` might result in undefined behavior. Consult IBM's documentation for details. Installations by a non-root user won't share installation data with the rest of the system.
+
+Default value: 'root'
+
+##### `options`
+
+Data type: `Any`
+
+Specifies options to pass to the installer.
+Default: - administrator mode: `-acceptLicense -s -log/tmp/IM_install.${timestamp}.log.xml -installationDirectory ${target}`. - nonadministrator/group mode: `-acceptLicense -accessRights nonAdmin -s -log /tmp/IM_install.${timestamp}.log.xml`
+
+Default value: `undef`
+
+##### `timeout`
+
+Data type: `Any`
+
+Specifies the timeout for the installation, in seconds. Defaults to 900. Installation Manager can take a long time to install, so if you have issues with Puppet timing out before the installation is complete, you might need to increase this parameter.
+
+Default value: '900'
+
+##### `installation_mode`
+
+Data type: `Any`
+
+Specifies which 'installation mode' you want to use to install the IBM Installation Manager. Values: 'administrator', 'nonadministrator', 'group'. Default: 'administrator'
+
+Default value: 'administrator'
+
+## Resource types
+
+### ibm_pkg
+
+Custom type for installing an IBM package.
+
+#### Properties
+
+The following properties are available in the `ibm_pkg` type.
+
+##### `ensure`
+
+Valid values: present, absent
+
+The basic property that the resource should be in.
+
+Default value: present
+
+#### Parameters
+
+The following parameters are available in the `ibm_pkg` type.
+
+##### `name`
+
+namevar
+
```

Comment:
Extra new lines.

Reply:
this is because "name" didn't have a `desc`. It does now.